### ISSUE TITLE
Suggestion: Add revisions to user-editable models that don't have them

### ISSUE THREAD

**Issue by elichad** [carpentries/amy#2344]
Notably, models in the `communityroles` and `extrequests` folders don't have versions/revisions. There may be others that I didn't find on a cursory browse of the codebase.

Community roles should probably have them.
We may not need them for workshop submissions/requests/inquiries as I'm not sure if these are regularly edited by CT members.

Generally, I would suggest that anything that has an Edit button in the AMY admin UI should have revisions enabled.

**Comment by maneesha** [comment]
I agree that anything that can be edited should have revisions enabled.
We don't often edit the workshop forms, but that is a good example of where edit history would be very useful, as it lets us see if a CT member made changes from the workshop host's original submission.
